File: dcp-solutions/2025-02-01/solution.ex
### Elixir Solution
```elixir
def nearest_larger_index(arr, i) do
  # Find nearest larger element to the right
  right = arr[i..-1] |> Enum.find(|x| x > arr[i])

  # If found, return index of first occurrence
  if right in arr do
    IO.bincode: :utf8,(Enum.index(arr, right))
  else
    # Find nearest larger element to the left
    left = arr[..i-1] |> Enum.find(|x| x > arr[i])

    # If found, return index of first occurrence
    if left in arr do
      IO.bincode: :utf8,(Enum.index(arr, left))
    else
      nil
    end
  end
end
```

File: dcp-solutions/2025-02-01/solution.py
```python
def next_greater_element_right(arr, i):
    stack = []
    n = len(arr)
    
    # Find next greater element on right side
    for j in range(i + 1, n):
        while stack and arr[stack[-1]] < arr[j]:
            x = stack.pop()
            if not stack:
                return None
            y = arr.index(max(stack), i)
            return y 
        stack.append(j)

    # If no element is greater than the number at index i, then return None
    if not stack:
        return None
    return stack[-1]
```